.github/workflows/tests.yml

```
@@ -0,0 +1,33 @@
+name: Unit Tests
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Set up shell environment
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y shellcheck
+        shell: bash
+      - name: Run shellcheck
+        run: shellcheck --severity warning scripts/upgrade.sh scripts/tests.sh
+
+  test:
+    name: Run Unit Tests
+    runs-on: ubuntu-latest
+    container:
+      image: busybox
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
```

Any particular reason why we use v1 while the shellcheck uses v3?
```suggestion
        uses: actions/checkout@v3
```

---

v2 and v3 uses node to invoke the checkout action https://github.com/actions/checkout/issues/334#issuecomment-727146723